frontend/pkg/frontend/frontend_test.go

```diff
@@ -219,12 +238,39 @@ func TestSubscriptionsPUT(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			if test.subDoc != nil {
-				err := f.dbClient.CreateSubscriptionDoc(context.TODO(), test.subDoc)
-				if err != nil {
-					t.Fatal(err)
+			// MiddlewareLockSubscription
+			// (except when MiddlewareValidateStatic fails)
+			mockDBClient.EXPECT().
+				GetLockClient().
+				MaxTimes(1)
+			if test.expectedStatusCode != http.StatusBadRequest {
+				// ArmSubscriptionPut
+				mockDBClient.EXPECT().
+					GetSubscriptionDoc(gomock.Any(), gomock.Any()).
+					DoAndReturn(func(context.Context, string) (*database.SubscriptionDocument, error) {
+						if test.subDoc != nil {
+							return test.subDoc, nil
+						} else {
+							return nil, database.ErrNotFound
+						}
+					})
+				// ArmSubscriptionPut
+				if test.subDoc == nil {
+					mockDBClient.EXPECT().
+						CreateSubscriptionDoc(gomock.Any(), gomock.Any())
+				} else {
+					mockDBClient.EXPECT().
+						UpdateSubscriptionDoc(gomock.Any(), gomock.Any(), gomock.Any())
 				}
 			}
+			// MiddlewareMetrics
+			// (except when MiddlewareValidateStatic fails)
+			mockDBClient.EXPECT().
+				GetSubscriptionDoc(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(ctx context.Context, subscriptionID string) (*database.SubscriptionDocument, error) {
+					return database.NewSubscriptionDocument(subscriptionID, test.subscription), nil
+				}).
```

Comment:
We can save putting a `Do` function call on the stack by just using [Return](https://pkg.go.dev/github.com/golang/mock/gomock#Call.Return) here.